pkgs/build-support/cc-wrapper/add-flags.sh

```diff
@@ -33,15 +41,27 @@ NIX_@infixSalt@_CFLAGS_COMPILE="-B@out@/bin/ $NIX_@infixSalt@_CFLAGS_COMPILE"
 # the script.
 
 if [ -e @out@/nix-support/libc-cflags ]; then
-    NIX_@infixSalt@_CFLAGS_COMPILE="$(< @out@/nix-support/libc-cflags) $NIX_@infixSalt@_CFLAGS_COMPILE"
+    NIX_@infixSalt@_CSTDLIB_COMPILE+=" $(< @out@/nix-support/libc-cflags)"
+fi
+
+if [ -e @out@/nix-support/libcxx-cxxflags ]; then
+    NIX_@infixSalt@_CXXSTDLIB_COMPILE+=" $(< @out@/nix-support/libcxx-cxxflags)"
+fi
+
+if [ -e @out@/nix-support/libcxx-ldflags ]; then
+    NIX_@infixSalt@_CXXSTDLIB_LINK+=" $(< @out@/nix-support/libcxx-ldflags)"
 fi
 
 if [ -e @out@/nix-support/cc-cflags ]; then
     NIX_@infixSalt@_CFLAGS_COMPILE="$(< @out@/nix-support/cc-cflags) $NIX_@infixSalt@_CFLAGS_COMPILE"
 fi
 
+if [ -e @out@/nix-support/cc-cxxflags ]; then
+    NIX_@infixSalt@_CFLAGS_COMPILE="$(< @out@/nix-support/cc-cxxflags) $NIX_@infixSalt@_CFLAGS_COMPILE"
```

Comment:
Why does this end up in CFLAGS instead of CXXFLAGS?

Reply:
Good point, though I am not sure that file is made by anything just yet. Historically we didn't have a `CXXFLAGS_COMPILE` and I am not sure whether I should be adding one in this, actually. Seems like the right thing to do but maybe later so we have 1 less moving part.

I originally thought we could do `CXXSTDLIB` -> `CXXFLAGS`, but that is not actually the case as we have that tricky logic around `-nostdinc`.

Reply:
There no longer is a `NIX_CXXFLAGS`